### PR TITLE
Add details to error message for junit

### LINKED ISSUE #7385: junit error message is not very descriptive
Currently when a test fails on setup or teardown, the junit error stored has a generic message indicating whether it was during setup or teardown. See https://github.com/pytest-dev/pytest/blob/master/src/_pytest/junitxml.py#L240.

This causes tools reading xml having to rely on the full trace to understand the error cause, which is not always easy to read as it can be a long message.

Can a descriptive short detail be added? I'll say something like

```python
 msg = "test teardown failure with error {}".format(report.longrepr.reprcrash.message)
```

That's a big simplification, all the [considerations](https://github.com/pytest-dev/pytest/blob/master/src/_pytest/junitxml.py#L219) taken with failures should be added as well

### PR DESCRIPTION
**Pull request by dajose**
closes  #7385

### CODE REVIEW

File: changelog/7385.improvement.rst [hunk 1]
```diff
@@ -0,0 +1 @@
+Add details to error message for junit.
```
Comment (nicoddemus) [inline changelog/7385.improvement.rst:1]:
```suggestion
``--junitxml`` now includes the exception cause in the ``message`` XML attribute for failures during setup and teardown. 

Previously:

.. code-block:: xml

    <error message="test setup failure">
    
Now:

.. code-block:: xml

    <error message="failed on setup with &quot;ValueError: Some error during setup&quot;">    
```

**Comment by nicoddemus** [comment]
(We should squash-merge this)

**Comment by dajose** [comment]
thanks @nicoddemus :) :) 

I'm already on the authors [:nerd_face:](https://github.com/pytest-dev/pytest/blob/master/AUTHORS#L78)

File: src/_pytest/junitxml.py [hunk 2]
```diff
@@ -236,10 +236,16 @@ def append_collect_skipped(self, report: TestReport) -> None:
         self._add_simple(Junit.skipped, "collection skipped", report.longrepr)
 
     def append_error(self, report: TestReport) -> None:
+        assert report.longrepr is not None
+        if getattr(report.longrepr, "reprcrash", None) is not None:
+            reason = report.longrepr.reprcrash.message
+        else:
+            reason = str(report.longrepr)
+
```
Comment (dajose) [inline src/_pytest/junitxml.py:244]: should I do?

```
reason = bin_xml_escape(reason)
```

**Comment by nicoddemus** [comment]
> I'm already on the authors 

Oh sorry, great then! 😁

**Comment by dajose** [comment]
@nicoddemus excuse me being intense... but are we waiting for more reviewers? or maybe something on my side?

**Comment by nicoddemus** [comment]
> excuse me being intense

Not at all, don't worry about pinging!

> but are we waiting for more reviewers? or maybe something on my side?

The former. We usually wait a bit to see if other maintainers want to take a second look.

I will merge this tomorrow unless someone else wants more time to review this. 👍

**Comment by nicoddemus** [comment]
Thanks @dajose!

**Comment by dajose** [comment]
thanks to you @nicoddemus  :)

**Pull request merged**